pkg/vmsock/conn_linux.go

```diff
@@ -88,13 +88,19 @@ func handleUDP(uConn *net.UDPConn, addr *net.UDPAddr, b []byte) {
 		return
 	}
 	data := make([]byte, UDPMaxBufer)
-	_, err = conn.Read(data)
+	n, err := conn.Read(data)
```

Comment:
Given that the data has a two byte length prefix, shouldn't we check that `n` is at least `data[0:1]` bytes long?

Reply:
Hmm, but this will also mean we need to sanity-check it (though that's 64k max, so it should be fine… other than the fact that the other end is UDP, so we need to cap it at 512).

Reply:
> the other end is UDP, so we need to cap it at 512

Isn't that an application layer concern. The tunnel should just pass through the data as it receives it (and it should already be truncated by the DNS server). The `vmsock` code shouldn't know anything about DNS internals.

Reply:
No, because our upstream (vsock) is a stream-based DNS server, and doesn't have the 512 byte limit.  And upstream of that is some syscalls (well, Win32 API calls), and has no limit either.

Reply:
This is wrong; the transport layer should not have any knowledge about the application layer protocols.

Now it doesn't matter right now, since we make sure the packets are small enough, but conceptually it still looks wrong to me.